azconfig/client/src/main/java/com/azure/azconfig/AzConfigCredentialsPolicy.java

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.azconfig;
+
+import com.microsoft.rest.v3.http.HttpHeaders;
+import com.microsoft.rest.v3.http.HttpPipelineCallContext;
+import com.microsoft.rest.v3.http.HttpPipelineLogLevel;
+import com.microsoft.rest.v3.http.HttpPipelineOptions;
+import com.microsoft.rest.v3.http.HttpRequest;
+import com.microsoft.rest.v3.http.HttpResponse;
+import com.microsoft.rest.v3.http.NextPolicy;
+import com.microsoft.rest.v3.http.policy.HttpPipelinePolicy;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import reactor.core.publisher.Mono;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.Base64;
+
+/**
+ * Creates a policy that authenticates request with AzConfig service.
+ */
+public final class AzConfigCredentialsPolicy implements HttpPipelinePolicy {
```

Comment:
This is a public class inside a public API package that does not have a public constructor. We need to decide if this class should become package-private, or if it should have a public constructor.